### PR TITLE
Added 3MF importer

### PR DESCRIPTION
**Pull request by gorilux**
Adds 3MF file format importer. 
(I'm still working on the exporter)

### CODE REVIEW

File: code/D3MFImporter.cpp [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+            {
+                return true;
+            }
+            else if (xmlReader->getNodeType() == irr::io::EXN_ELEMENT_END &&
+                     xmlReader->getNodeName() == startTag)
+            {
+                return false;
+            }
+        }
+        //DefaultLogger::get()->error("unexpected EOF, expected closing <" + closeTag + "> tag");
```
Comment (kimkulling) [inline code/D3MFImporter.cpp]: Dead code.

**Comment by kimkulling** [comment]
Great work, this format was one of my next candidates! Thanks a lot for this great contribution.

**Comment by acgessler** [comment]
Awesome!

**Comment by gorilux** [comment]
Txs! For accepting pull request. In the next couple of weeks I'm hoping to upload the exporter.

**Pull request merged**